src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = "supply";
+    private static final String BUY_LABEL = "buy";
+    private static final String RESULT_LABEL = "result";
+    private static final String DELIMITER = ",";
+    private static final String NEW_LINE = "\n";
 
+    private int[] readAndAggregate(String fromFileName) {
+        int[] result = new int[2];
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line = reader.readLine();
+            String[] key = new String[2];
+            while (line != null) {
+                if (key.length == 2) {
+                    key = line.split(DELIMITER);
+                    if (key[0].equals(SUPPLY_LABEL)) {
+                        result[0] += Integer.parseInt(key[1]);
+                        line = reader.readLine();
+                    } else if (key[0].equals(BUY_LABEL)) {
+                        result[1] += Integer.parseInt(key[1]);
+                        line = reader.readLine();
+                    } else {
+                        throw new RuntimeException("Wrong word " + line + "in file "
+                                + fromFileName);
```

Operation handling treats anything other than exact matches of `key[0]` to supply/buy as an error but the code's current malformed-line checks are ineffective. Also the explicit error messages are poorly formatted (missing spaces) and lack full context.

This relates to checklist guidance: "Don't ignore exceptions" and the task requirement to validate operations explicitly. Update to trim and validate `op = tokens[0].trim()` and then `if (SUPPLY_LABEL.equals(op)) ... else if (BUY_LABEL.equals(op)) ... else { throw new RuntimeException("Wrong operation '" + op + "' in file " + fromFileName + " for line '" + line + "'"); }`. Also fix message formatting (e.g. currently "Wrong word " + line + "in file ").